frame/contracts/src/wasm/runtime.rs

```diff
@@ -717,9 +735,25 @@ where
 		if value > 0u32.into() {
 			self.charge_gas(RuntimeCosts::CallSurchargeTransfer)?;
 		}
-		let ext = &mut self.ext;
-		let call_outcome =
-			ext.call(gas, callee, value, input_data, flags.contains(CallFlags::ALLOW_REENTRY));
+
+		let call_outcome = match callee {
+			#[cfg(feature = "unstable-interface")]
+			CallType::CallCode(code_hash_ptr) => {
+				let code_hash = self.read_sandbox_memory_as(code_hash_ptr)?;
+				self.ext.call_code(code_hash, input_data)
+			},
+			CallType::Call(callee_ptr) => {
+				let callee: <<E as Ext>::T as frame_system::Config>::AccountId =
+					self.read_sandbox_memory_as(callee_ptr)?;
+				self.ext.call(
+					gas,
+					callee,
+					value,
+					input_data,
+					flags.contains(CallFlags::ALLOW_REENTRY),
```

Comment:
Please check that this flag isn't set in case of a `CallCode` and throw a new `InvalidCallFlags` module error. This error should also replace the adhoc string error of "used reserved bit in CallFlags".